Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,149 @@
+
+        public const string DiagnosticListenerName = "SqlClientDiagnosticListener";
+
+        private const string SqlClientPrefix = "System.Data.SqlClient.";
+
+        public const string SqlAfterExecuteCommand = SqlClientPrefix + "WriteCommandAfter";
+        public const string SqlAfterOpenConnection = SqlClientPrefix + "WriteConnectionOpenAfter";
+
+        private readonly TelemetryClient client;
+        private readonly SqlClientDiagnosticSourceSubscriber subscriber;
+
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher startTimeFetcher = new PropertyFetcher("StartTime");
+        private readonly PropertyFetcher durationFetcher = new PropertyFetcher("Duration");
+
+        public SqlClientDiagnosticSourceListener(TelemetryConfiguration configuration)
+        {
+            this.client = new TelemetryClient(configuration);
+            this.client.Context.GetInternalContext().SdkVersion =
+                SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceCore + ":");
+
+            this.subscriber = new SqlClientDiagnosticSourceSubscriber(this);
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnCompleted()
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnError(Exception error)
+        {
+        }
+
+        void IObserver<KeyValuePair<string, object>>.OnNext(KeyValuePair<string, object> evnt)
+        {
+            switch (evnt.Key)
+            {
+                case SqlAfterOpenConnection:
+                {
+                    this.OnAfterOpenConnection(
+                        (SqlConnection) connectionFetcher.Fetch(evnt.Value),
+                        (DateTimeOffset) this.startTimeFetcher.Fetch(evnt.Value),
+                        (TimeSpan) this.durationFetcher.Fetch(evnt.Value));
+
+                    break;
+                }
+            }
+        }
+
+        public const string SqlConnectionCommandText = "Open";
+
+        private void OnAfterOpenConnection(SqlConnection connection, DateTimeOffset timestamp, TimeSpan duration)
+        {
+            var telemetry = new DependencyTelemetry();
+
+            // TODO: Activities?
+
+            this.client.Initialize(telemetry); // TODO: Call this?
+
+            // TODO: Unify with ProfilerSqlConnectionProcessing?
+            telemetry.Name = string.Join(" | ", connection.DataSource, connection.Database, SqlConnectionCommandText);
+            telemetry.Type = RemoteDependencyConstants.SQL;
+            telemetry.Target = string.Join(" | ", connection.DataSource, connection.Database);
+            telemetry.Data = SqlConnectionCommandText;
+            telemetry.Timestamp = timestamp;
+            telemetry.Duration = duration;
+
+            this.client.Track(telemetry);
```

Comment:
Yes. Once we are happy with the general approach I will implement the rest. Take a look [here](https://github.com/dotnet/corefx/blob/master/src/System.Data.SqlClient/src/System/Data/SqlClient/SqlClientDiagnosticListenerExtensions.cs)for what is available.